### PR TITLE
Chapter 3

### PR DESCRIPTION
**Pull request by NurAnisAqilah**

### CODE REVIEW

**Comment by NurAnisAqilah** [comment]
Ready

File: .gitignore [hunk 1]
```diff
@@ -32,10 +32,13 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline .gitignore]: Please fix conflict in this file

File: .ruby-version [hunk 2]
```diff
@@ -1 +1,5 @@
+<<<<<<< HEAD
 2.6.5
+=======
+2.7.1
+>>>>>>> 4b25ebe... Chapter 3
```
Comment (chungpv-1008) [inline .ruby-version]: similar

File: Gemfile [hunk 3]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline Gemfile]: similar

File: Gemfile.lock [hunk 4]
```diff
@@ -85,7 +85,11 @@ GEM
     listen (3.2.1)
       rb-fsevent (~> 0.10, >= 0.10.3)
       rb-inotify (~> 0.9, >= 0.9.10)
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline Gemfile.lock]: similar

File: app/javascript/packs/application.js [hunk 5]
```diff
@@ -1,8 +1,24 @@
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline app/javascript/packs/application.js]: similar

File: app/mailers/application_mailer.rb [hunk 6]
```diff
@@ -1,4 +1,9 @@
 class ApplicationMailer < ActionMailer::Base
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline app/mailers/application_mailer.rb]: similar

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -5,8 +5,13 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline app/views/layouts/application.html.erb]: similar

File: bin/rails [hunk 8]
```diff
@@ -1,9 +1,12 @@
 #!/usr/bin/env ruby
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline bin/rails]: similar

File: bin/rake [hunk 9]
```diff
@@ -1,9 +1,12 @@
 #!/usr/bin/env ruby
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline bin/rake]: similar

File: config/credentials.yml.enc [hunk 10]
```diff
@@ -1 +1,5 @@
-EG7slYG86cPFf9TUEDK2oSMn1djrkhHAEC0H8XmriCbnBIGBBEDdUeRkezzSutBmfjc0PdRdCwq7waqpPvucA/XjaWlClfxvnBpeS4aYRE7WeGJylUDkpmoJ1Nl/UEXmFAsVpaAqFEPT58DwuYoYWtgh3KVKkyQMyfDOP8ZeUOPTzjiGCcpteuoQwP4uX/U8TcivO6dt7OQYAcweh9z0VpWTYynGqOSpZUlqoU43CdnerN7KULy8RpEMPbjKr0E+95qQbMObK4n/k6+JfYzH+qhH8k5ZJwnSKlah1ccJb5Yd4nxqhgveTFSNHvu3fmXOr3nBHIuldwbhcg1VA2N4uHuzVsQ7sD3TGO1Z8c/i+SmZ72yjeKs9vm5c1N2UhsnVGzLccCi3+JwPp7+hupF1soAvsES0aTHrqPWt--+wLtIgFWsx7S75Ti--0+SJkw2DgcBL0iXIQ+/Idg==
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline config/credentials.yml.enc]: similar

File: config/environments/test.rb [hunk 11]
```diff
@@ -6,8 +6,12 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
+<<<<<<< HEAD
```
Comment (chungpv-1008) [inline config/environments/test.rb]: similar

File: .gitignore (hunk 1 above)
Comment (saobv-2223) [inline .gitignore]: Please, resolve conflict. There are many errors in this pull request

**Review by saobv-2223 — COMMENTED** [review]
commented

**Comment by vanvtt-0952** [comment]
commented

**Comment by NurAnisAqilah** [comment]
ready

File: .gitignore (hunk 1 above)
Comment (quoctd-0607) [reply .gitignore]: remove

File: .gitignore [hunk 12]
```diff
@@ -32,10 +32,13 @@
...
 
 /public/packs
 /public/packs-test
 /node_modules
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
+=======
+>>>>>>> 4b25ebe... Chapter 3
```
Comment (quoctd-0607) [inline .gitignore]: remove

File: Gemfile [hunk 13]
```diff
@@ -43,3 +44,56 @@ end
 
 
 gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+=======
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '2.7.1'
```
Comment (quoctd-0607) [inline Gemfile]: using double quote

**Comment by NurAnisAqilah** [comment]
ready

**Comment by manhnt-0960** [comment]
commented
please fix 5 commits

**Comment by NurAnisAqilah** [comment]
ready

**Comment by NurAnisAqilah** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented
conflict code

**Comment by NurAnisAqilah** [comment]
ready

File: Gemfile [hunk 14]
```diff
@@ -1,23 +1,29 @@
 source "https://rubygems.org"
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby "2.6.5"
+
+ruby "2.7.1"
+
```
Comment (saobv-2223) [inline Gemfile:5]: remove this line

File: app/views/layouts/application.html.erb [hunk 15]
```diff
@@ -10,6 +10,10 @@
   </head>
 
   <body>
+  <%= render "layouts/header" %>
+  <div class="container">
     <%= yield %>
-  </body>
+  </div>
+
+</body>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]:
```suggestion
  </div>
</body>
```

File: config/credentials.yml.enc [hunk 16]
```diff
@@ -1 +1 @@
-EG7slYG86cPFf9TUEDK2oSMn1djrkhHAEC0H8XmriCbnBIGBBEDdUeRkezzSutBmfjc0PdRdCwq7waqpPvucA/XjaWlClfxvnBpeS4aYRE7WeGJylUDkpmoJ1Nl/UEXmFAsVpaAqFEPT58DwuYoYWtgh3KVKkyQMyfDOP8ZeUOPTzjiGCcpteuoQwP4uX/U8TcivO6dt7OQYAcweh9z0VpWTYynGqOSpZUlqoU43CdnerN7KULy8RpEMPbjKr0E+95qQbMObK4n/k6+JfYzH+qhH8k5ZJwnSKlah1ccJb5Yd4nxqhgveTFSNHvu3fmXOr3nBHIuldwbhcg1VA2N4uHuzVsQ7sD3TGO1Z8c/i+SmZ72yjeKs9vm5c1N2UhsnVGzLccCi3+JwPp7+hupF1soAvsES0aTHrqPWt--+wLtIgFWsx7S75Ti--0+SJkw2DgcBL0iXIQ+/Idg==
\ No newline at end of file
+Wcfs5r7nZwIW/5hXDT005dm0mAZ40stI6NJ2oQr2/DF8AWGYH+U+zsWIKd7BXqCegLR9dYXR2X7uIDORVLb+C8jLqB8n79WyEn7Dwv85vqALatfVr1QfMMbOwDy8kZXtf2wX+pBu3NHqL6/71iVwE7ilHMPy4GiRJW7duPEtxlZykhQZTELzrzbJ2SQFDdilzw5NTMXrd+rqekm6BkAgk/jSFBqyAryZBn8LI0emmAQgCe0B62+qD9wdGU8fagW8tI/qIbx6IHXscRly5Bt4r5VLP7suVE2dWXxvcUP/9dmsPqBZwNIY5EeM8B7QeEA7ZNZqUzF193odBES9o/HWvU24Gu6/RcH8Pz4a9VyoLN1uS+c6b2GzSB5umVbD41DqSVpvRYjGebzCaaT87aXf1hv/NVE7p3eMsiP4--ysumdpMTx5V2TXl3--KQtAHPmqb+la5vUbJt+O/A==
```
Comment (saobv-2223) [inline config/credentials.yml.enc]: image

File: config/database.yml [hunk 17]
```diff
@@ -2,7 +2,7 @@
 #   gem install sqlite3
 #
 #   Ensure the SQLite 3 gem is defined in your Gemfile
-#   gem 'sqlite3'
+#   gem "sqlite3"
```
Comment (saobv-2223) [inline config/database.yml]: image

File: config/settings.yml [hunk 18]
```diff
@@ -0,0 +1,6 @@
+links:
+  rails_tutorial:
+    help_page: http://www.railstutorial.org/help
+    book_page: https://www.railstutorial.org/book
+    home_page: http://www.railstutorial.org/
+
```
Comment (saobv-2223) [inline config/settings.yml]: add endline

**Comment by NurAnisAqilah** [comment]
ready

File: .gitignore [hunk 19]
```diff
@@ -39,3 +39,6 @@
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
+/config/credentials.yml.enc
+/config/database.yml
+/test
```
Comment (saobv-2223) [inline .gitignore]:
```suggestion
/test

```

**Review by saobv-2223 — COMMENTED** [review]
image

**Comment by NurAnisAqilah** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
merged

**Pull request merged**